### ISSUE TITLE
Enhancement: Ability to specify the preview image in the content type

### ISSUE THREAD

**Issue by estruyf** [estruyf/vscode-front-matter#124]
Specify which field can be used for the preview image in the content type.

The issue raised by @appieschot on Twitter: https://twitter.com/appieschot/status/1443568883248492554?s=20

**Comment by estruyf** [comment]
This will be supported in the upcoming release where you can specify in your content type image field that it can be used as the preview image with the `isPreviewImage` property.

```json
{
  "title": "Article preview",
  "name": "custom-image",
  "type": "image",
  "isPreviewImage": true
}
```

**Comment by appieschot** [comment]
Awesome work 🚀, two more questions though ;-): 
- My images are stored in the /static folder (Hugo specific). When referencing you dont specify that folder and Hugo knows (https://gohugo.io/content-management/static-files/) any way we can configure that for preview's as well? 
- Even when specifying the preview URL correctly the image won't show up; it does show in the left pane as preview, but the dashboard shows an error image. Is there a place where I can look for logs?

**Comment by estruyf** [comment]
@appieschot did you specify the `frontmatter.content.publicfolder` setting with `static` as value?

https://frontmatter.codes/docs/settings#frontmatter.content.publicfolder

**Comment by appieschot** [comment]
Yes, the preview itself shows on the left pane (see screenshot) but throws a image not found in the dashboard.

![image](https://user-images.githubusercontent.com/15227781/135603822-e5ea4cb9-d413-4428-8f16-c9ad5d3bce0e.png)

Part of the settings: 

```
 "frontMatter.taxonomy.contentTypes": [
        {
            "name": "default",
            "fields": [
                {
                    "title": "Title",
                    "name": "title",
                    "type": "string"
                },
                {
                    "title": "Publishing date",
                    "name": "date",
                    "type": "datetime"
                },
                {
                    "title": "Article preview",
                    "name": "image",
                    "type": "image",
                    "isPreviewImage": true
                },
                {
                    "title": "Is in draft",
                    "name": "draft",
                    "type": "boolean"
                },
                {
                    "title": "Categories",
                    "name": "categories",
                    "type": "categories"
                }
            ]
        }
    ],
    "frontMatter.content.pageFolders": [
        {
            "title": "post content",
            "path": "[[workspace]]/content"
        }
    ],
    "frontMatter.content.publicFolder": "static",
    "frontMatter.content.autoUpdateDate": true
```

**Comment by estruyf** [comment]
Would you be able to open the developer tools and check which URL of the image it tries to load?

**Comment by appieschot** [comment]
Learned something new; didn't know it would be that easy. 

Image on the left is rendered using: 
```<img src="https://file%2B.vscode-resource.vscode-webview.net/c%3A/git/sbx/weblog/static/wp-content/uploads/2021/cliflowexport.webp">```

The image in the dashboard is: 
```<img src="/wp-content/uploads/2021/cliflowexport.webp" alt="Flow missing export to logic app? Use the CLI for Microsoft 365" class="absolute inset-0 h-full w-full object-cover" loading="lazy">```

Manually changing the URL only solves the problem.

**Comment by estruyf** [comment]
ok, know what the issue is 😀, will fix it

**Comment by appieschot** [comment]
Will have a look at the code to see if I can help out next time as well 💪

**Comment by appieschot** [comment]
After updating the panel won't load anymore, anything else I need to update / check / change? 

![image](https://user-images.githubusercontent.com/15227781/135635167-c0ae98d5-0938-41d2-a3b5-6578261c8391.png)

**Comment by estruyf** [comment]
Seems something went wrong during the build

**Comment by estruyf** [comment]
Ok, found it. Auto-import inserted something in the wrong order. Should be fixed in the next release which is currently publishing.

**Comment by estruyf** [comment]
Released and works again @appieschot 😁

**Comment by appieschot** [comment]
Awesome works great! Can be closed out 🚀
